web/src/layers/EditableIntersectionLayer.svelte

```diff
@@ -0,0 +1,97 @@
+<script lang="ts">
+  import { getContext } from "svelte";
+  import {
+    CircleLayer,
+    hoverStateFilter,
+    Popup,
+    SymbolLayer,
+  } from "svelte-maplibre";
+  import { layerId, mapMetersToPixels } from "../common";
+  import { Style } from "../common/colors";
+  import {
+    Intersection,
+    type IntersectionFeature,
+  } from "../common/Intersection";
+  import EditIntersectionPopup from "../edit/EditIntersectionPopup.svelte";
+  import type { RenderNeighbourhoodOutput } from "../wasm";
+
+  export let neighbourhood: RenderNeighbourhoodOutput =
+    getContext("neighbourhoodGj");
+  export let onClickIntersection = (intersection: Intersection) => {};
+  export let interactive: boolean = false;
+
+  /// NOTE: this takes the intersection's index in the neighborhood FeatureCollection, *not* the intersectionId!
+  function getIntersectionByFeatureIndex(
+    intersectionIdx: number,
+  ): Intersection {
+    // The Feature from maplibre has its nested properties serialized as a JSON string.
+    // see: https://github.com/dimfeld/svelte-maplibre/discussions/162#discussioncomment-9295264
+    //
+    // So we refetch the corresponding IntersectionFeature from the neighbourhood to get all it's properties.
+    let intersectionFeature = neighbourhood.features[intersectionIdx];
+    let intersection = new Intersection(
+      intersectionFeature as IntersectionFeature,
+    );
+    return intersection;
+  }
+
+  function getIntersectionFromFeatures(features: any): Intersection {
+    // Note: this `id` is autoassigned by maplibre, and corresponds to the index of the feature in the neighbourhood.
+    return getIntersectionByFeatureIndex(features[0].id);
+  }
+</script>
+
+<!-- 
+ FIXME: If the user clicks twice to add and then immediately rotate a filter, they will be zoomed in.
+ I haven't looked too deeply int how maplibre manages this, but none of the following prevents zooming...
+ 
+     event.preventDefault();
+     event.stopPropagation();
+     event.stopImmediatePropagation();
+
+ Maybe because `event.cancelable == false`
+-->
+<CircleLayer
+  {...layerId("editable-intersections")}
+  filter={["==", ["get", "kind"], "editable_intersection"]}
+  paint={{
+    "circle-radius": mapMetersToPixels(30),
+    "circle-color": Style.mapFeature.hover.backgroundColor,
+    "circle-opacity": hoverStateFilter(0.0, 0.5),
+    "circle-stroke-color": Style.mapFeature.hover.backgroundColor,
+    "circle-stroke-opacity": 0.8,
+    "circle-stroke-width": hoverStateFilter(0, 3),
+  }}
+  {interactive}
+  manageHoverState
+  hoverCursor="pointer"
+  on:click={(event) =>
+    onClickIntersection(getIntersectionFromFeatures(event.detail.features))}
+>
+  <Popup openOn="hover" let:features>
+    <EditIntersectionPopup
+      intersection={getIntersectionFromFeatures(features)}
+    />
+  </Popup>
+</CircleLayer>
+
+<SymbolLayer
```

Comment:
This is the part that's confusing me. This component is doing two jobs, letting people interactively click intersections and rendering diagonal filters. For route mode and impact modes, we only want to render diagonal filters -- and in fact, we want to show them over the entire map, not just one neighbourhood.

What do you think about making a new map-wide API call to render all diagonal filters, or folding it into `renderModalFilters`?

Reply:
I think I understand your concern.

To be clear, unless I'm mistaken, the layer is only editable when you're in `Editing` mode. Everywhere else it has `interactive={false}`, which is sufficient to make it readonly (unless there's a bug....).

But two things I need to fix:

1. `EditableIntersectionLayer` is a terrible name for a layer that's often not editable.
2. I want these to be visible whenever the modalFilters are visible - e.g. in the "pick-neighbourhood" mode, which it currently does not. I'll take a look at folding it into renderModalFilters.

Reply:
Ok, I've pushed up some changes and I think this is as  you described now (which is much better!) 

"Some day" I'd like to introduce some kind of `struct SaveFile`. I thinking working with geojson::Features is making it a bit clumsy.